src/django_idom/components.py

```diff
@@ -1,9 +1,121 @@
+from __future__ import annotations
+
 import os
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Dict, Iterable
 
+from channels.db import database_sync_to_async
 from django.contrib.staticfiles.finders import find
-from idom import component, html
+from django.http import HttpRequest
+from django.urls import reverse
+from django.views import View
+from idom import component, hooks, html, utils
+from idom.types import VdomDict
+
+from django_idom.config import IDOM_CACHE, IDOM_VIEW_COMPONENT_IFRAMES
+from django_idom.types import ViewComponentIframe
+
+
+# TODO: Might want to intercept href clicks and form submit events.
+# Form events will probably be accomplished through the upcoming DjangoForm.
+@component
+def view_to_component(
+    view: Callable | View,
+    compatibility: bool = False,
+    transforms: Iterable[Callable[[VdomDict], Any]] = (),
+    strict_parsing: bool = True,
+    request: HttpRequest | None = None,
+    args: Iterable = (),
+    kwargs: Dict | None = None,
+) -> VdomDict | None:
+    """Converts a Django view to an IDOM component.
+
+    Args:
+        view: The view function or class to convert.
+
+    Keyword Args:
+        compatibility: If True, the component will be rendered in an iframe.
+            Strict parsing does not apply to compatibility mode.
+        transforms: A list of functions that transforms the newly generated VDOM.
+            The functions will be called on each VDOM node.
+        strict_parsing: If True, an exception will be generated if the HTML does not
+            perfectly adhere to HTML5.
+        request: Request object to provide to the view.
+        args: The positional arguments to pass to the view.
+        kwargs: The keyword arguments to pass to the view.
+    """
+    kwargs = kwargs or {}
+    rendered_view, set_rendered_view = hooks.use_state(None)
+
+    # Asynchronous view rendering via hooks
+    @hooks.use_effect
+    async def async_renderer():
+        """Render the view in an async hook to avoid blocking the main thread."""
+        # Avoid re-rendering the view
+        if rendered_view or compatibility:
```

Comment:
The problem with `args` and `kwargs` is that it's hard to tell if their values change.

Reply:
This is actually a bug I'm realizing. I would like the preserve the ability to re-render if `args` or `kwargs` change.

In the future, we'll also want to trigger a re-render when HTTP methods change (which is why it originally was outside of the `async_renderer`. 

For now, I will set `dependencies=[args, kwargs]`

Reply:
So the pattern here would be to do the following:

```python
def view_to_component(view, ..., dependencies):
    view = use_callback(view, dependencies=dependencies)
    rendered_view, set_rendered_view = use_state(None)

    @use_effect
    async def render_view():
        set_rendered_view(view())

    return rendered_view
```

Usage being:

```python
view_to_component(view)
view_to_component(lambda: view(...))
view_to_component(lambda: view(...), dependencies=...)
```

Reply:
> The problem with `args` and `kwargs` is that it's hard to tell if their values change.

Do you have a proposed solution for this?

Reply:
`dependencies=[args, kwargs]` won't quite work either since the user is probably doing:

```
view_to_component(view, args=(...), kwargs={...})
```

And the identity of `args` and `kwargs` will change every time. You could expand all the values within `args` and `kwargs`, but that get's pretty hacky.

The pattern I suggest above, of requiring the user to wrap functions that require more arguments, is pretty common in React. It has the benefit of allowing us to only check the values we know will change by leveraging the magic of `use_callback`.

Reply:
Wrapping the view with `lambda` completely murders the current rendering schema. Definitely going to bring significantly inefficiency surrounding async views.

Reply:
Is there any way that the logic could be re-worked to check `inspect.isawaitable` instead of `iscoroutinefunction`?

If not, perhaps you could restrict parameters to positional arguments? That's at least a little easier to destructure and pass to `dependencies` in the effect:

```python
def view_to_component(view, *args, ...):
    
    @use_effect(dependencies=[view, *args])
    async def render_view(): ...
```

Reply:
Actually, `isawaitable` doesn't seem viable. I'd go with the second option. Limit parameters to positional args.

Reply:
Or, if you really want to allow kwargs you'll have to do something like the following:

```python
dependencies = [..., *args] + [x for item in kwargs.items() for x in item]
```

Reply:
Unfortunately, kwargs are technically the more important thing here.

In theory, views should always be passed serializable data since the values [come from either URLs parts](https://docs.djangoproject.com/en/4.1/topics/http/urls/#example) or URL parameters. So maybe we can do `dependencies = [request, *args, json.dumps(kwargs)]`?

Reply:
Ah, that's a clever solution. You'll probably need to silence some errors from `flake8_idom_hooks` but this should be fine.

Reply:
Err, if you want to `json.dumps(kwargs)` you should do the same for `args` to keep the behavior consistent. De-structuring `args` into dependencies will make the effect check the identity of each value for changes. Whereas, with `json.dumps` effectively causes the effect to check for value changes.

Reply:
I'm thinking of doing the same for the request object, to avoid mutability.

Something similar to `json.dumps(vars(request_obj))`

Perhaps with `default=lambda x: None` to avoid callables from breaking things.

Reply:
I think the latest solution is the best we're going to get.

On a different note, `flake8-idom-hooks` identifies `rendered_view, set_rendered_view = hooks.use_state(None)` and `@hooks.use_effect` as `hook 'use_effect' used outside component or hook definition`

Reply:
I haven't released the latest `flake8-idom-hooks` yet, so it's not considering the `@component` decorator.

Reply:
I changed the component name to CamelCase for testing. Looks like it doesn't give any warnings, so no need to suppress anything.

Is the latest implementation good to go?

Reply:
Are there any tests that check whether the component will re-render appropriately if inputs change?

Reply:
I will make one today

Reply:
New tests are live

Reply:
Also just released the latest flake8 plugin.

Reply:
Will review the new tests on Sunday